lib/websocket-server.js

```diff
@@ -302,6 +302,8 @@ class WebSocketServer extends EventEmitter {
       }
     }
 
+    ws.url = req.url;
```

Comment:
`req.url` is only the the URL `path` so this is not acceptable.

Reply:
The problem is also getting the original URL `protocol` and `host` when server is behind a proxy. I don't think it's worth the added complexity that this will result in.

Users can already access `req.url`.

Reply:
That makes sense. A full solution is more trouble than it is worth. The simple `req.url` was enough for my use (behind a proxy). It seemed simple enough to add to the server code, but this would just be a different incompatibility between the WebSocket objects, not an improvement.